### ISSUE TITLE
custom-error-definition rule does not validate against exported name

### ISSUE THREAD

**Issue by MrHen** [sindresorhus/eslint-plugin-unicorn#190]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2440%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/190)

The following passes linting:

```js
exports.fooError = class FooError extends Error {
    constructor(error) {
        super(error);
        this.name = 'FooError';
    }
};
```

The name used in `exports.fooError` should match the name used in `this.name = 'FooError';`.

A commented out test was added as part of #188 to cover this scenario.

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by IssueHuntBot** [comment]
@issuehunt has funded $40.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/55832243/issues/190) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/55832243/issues/190).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by IssueHuntBot** [comment]
@sindresorhus has rewarded $36.00 to @mrhen. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/190)

- :moneybag: Total deposit: $40.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $4.00
